### DISCUSSION TITLE
Using virtual machine as slave machine

### DISCUSSION THREAD

**Discussion by bsjdsp** [apohl79/audiogridder#564] [General]
I like to mix on Linux but due to the lack of plugins and my uncomfortableness with windows and using it due to how much privacy is hindered by big corporations. Is it a good idea to have windows on a virtual machine to host the plugins to stream. I do have a workstation with decent specs. I’m just not sure if anyone has done this before

**Comment by dhstsw** [comment]
I'm doing it on a VM under KVM (unraid) with 6 cores assigned to it.
It works, but i'm still troubleshooting pops and glitches in the audio stream (i think due to some overhead on the virtual ETH drivers).

**Reply by jaacMusic** [reply]
The more layers you put on the AG software between the VM OS and the host OS it adds more latency as it has to conduct more translation from the VM layer to the host layer and so on. The best configuration is to have it running at the host level and strip as many services/apps from the OS as you can. This will allow more RAM and resources to be focused on the AG processing engine.

**Reply by dhstsw** [reply]
A VM under KVM has very little layers to the bare metal, since basically everything is passed directy (if you do it right). Infact you can do gaming on it without basically any difference in performance.
The only problem is that the only thing that matters for AG is the ethernet port. I only have one and on *my* motherboard i can't add a cheap ethernet port to dedicate (and pass through) to the VM.
So, it's not an universal isssue: if one can afford to dedicate an additional ethernet port to the VM there should be no problem.

**Comment by Fittkenny** [comment]
Hmmm...seems to me the virtual machine thingie is totally useless? What is it I don't get?

**Reply by dhstsw** [reply]
See above.

**Reply by Fittkenny** [reply]
You get clicks and pops ...gee I wonder why....use it on a dedicated pc and it will work.

**Reply by dhstsw** [reply]
Yup, i see you understood everything i wrote. :D

**Comment by Fittkenny** [comment]
It’s futile, it won’t work.

**Reply by dhstsw** [reply]
Yes, you're totally right.
Except for the fact that in the while i redefined the VM to use i440fx instead of Q35 and it does.
Flawlessy.

Cheers.

**Comment by naokiapri** [comment]
i'm using audiogridder on unraid server (win10 vm) with double xeons and a lot of ram — great extension for m1 macbook with only 8 gb ram. especially when using kontakt.
